src/audio/tdfb/tdfb.c

```diff
@@ -336,6 +512,64 @@ static int tdfb_cmd_get_data(struct comp_dev *dev,
 	return ret;
 }
 
+static int tdfb_cmd_switch_get(struct sof_ipc_ctrl_data *cdata, struct tdfb_comp_data *cd)
+{
+	int j;
+
+	/* Fail if wrong index in control, needed if several in same type */
+	if (cdata->index != CTRL_INDEX_PROCESS)
+		return -EINVAL;
+
+	for (j = 0; j < cdata->num_elems; j++)
+		cdata->chanv[j].value = cd->beam_on;
+
+	return 0;
+}
+
+static int tdfb_cmd_enum_get(struct sof_ipc_ctrl_data *cdata, struct tdfb_comp_data *cd)
+{
+	int j;
+
+	switch (cdata->index) {
+	case CTRL_INDEX_AZIMUTH:
+		for (j = 0; j < cdata->num_elems; j++)
+			cdata->chanv[j].value = cd->az_value;
+
+		break;
+	case CTRL_INDEX_AZIMUTH_ESTIMATE:
+		for (j = 0; j < cdata->num_elems; j++)
+			cdata->chanv[j].value = cd->az_value_estimate;
+
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int tdfb_cmd_get_value(struct comp_dev *dev, struct sof_ipc_ctrl_data *cdata)
+{
+	struct tdfb_comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_ENUM:
+		comp_dbg(dev, "tdfb_cmd_get_value(), SOF_CTRL_CMD_ENUM index=%d", cdata->index);
+		ret = tdfb_cmd_enum_get(cdata, cd);
+		break;
+	case SOF_CTRL_CMD_SWITCH:
+		comp_dbg(dev, "tdfb_cmd_get_value(), SOF_CTRL_CMD_SWITCH index=%d", cdata->index);
+		ret = tdfb_cmd_switch_get(cdata, cd);
+		break;
+	default:
+		comp_err(dev, "tdfb_cmd_get_value() error: invalid cdata->cmd");
+		ret = -EINVAL;
+		break;
```

Comment:
in fact you could just do `return ...` in each case and not bother with `ret` at all

Reply:
done